### DISCUSSION TITLE
Unexpected error when importing HashSet

### DISCUSSION THREAD

**Discussion by ornamentist** [rimbu-org/rimbu#150] [Q&A] [answered]
I'm probably doing something wrong here. I'm trying out Rimbu's HashSets in Typescript with bun:

```
import {HashSet} from "@rimbu/core"

const setFromValues = HashSet.of(1, 3, 2, 3);
const setFromIterable = HashSet.from("cdcba");
const emptySet = HashSet.empty<string>();

console.log(setFromValues)
console.log(setFromIterable)
console.log(emptySet)
```

This file compiles fine but I see the following error when I run it:

```
~> bun tsc
~> bun run src/sets/rset.ts 
1 | import { Token } from '@rimbu/base';
2 | import { CollectFun, OptLazy, TraverseState } from '@rimbu/common';
3 | import { fromStreamSource, isEmptyStreamSourceInstance, } from '@rimbu/stream/custom';
   ^
ReferenceError: Cannot access uninitialized variable.
      at /Users/stu/projects/mathe/TS/node_modules/@rimbu/stream/dist/module/custom/fast-iterator-custom.js:3:0
```

Any advice much appreciated.

**Comment by vitoke** [comment]
Thanks for reporting! It looks like you're not doing anything wrong. I tried your code snippet in a CodeSandbox instance, and that works fine.

I have not personally tried to use bun yet, but I suspect the issue lies there somewhere. I ran your example with bun and indeed get the same error message. However, if I run it with bun inside the rimbu code base after building it normally, it works fine. It is a mystery to me why that would be.

Since bun seems to be able to directly run TypeScript, it could be best to point bun to the TypeScript code instead of the transpiled code. I see in the bun documentation that they support the `exports` field in `package.json` with a custom `bun` conditional. I wanted to migrate to package exports anyway so I'll see if I can do that on the near term and see if that works.. Hopefully somewhere this week, but no promises :)

Created an [issue](https://github.com/rimbu-org/rimbu/issues/151)

**Reply by ornamentist** [reply]
I'll give that a go, thanks. I'm very grateful that Rimbu exists and is well supported.

**Reply by vitoke** [reply]
While trying to get `bun` to directly run the rimbu TypeScript code, I found out that `bun` seems to have an issue running some TypeScript code. This is currently preventing me from being able to run rimbu directly with `bun`. I have also created an [issue](https://github.com/oven-sh/bun/issues/2744) for this in the `bun` repo.

**Comment by vitoke** [comment]
Good news! I think I managed to get rimbu to work with `bun`. You will need to use the latest released rimbu versions (e.g. for `@rimbu/core` this is 0.11.5 atm).

Furthermore, as mentioned before a bug in `bun` is keeping the current released `bun` version (0.5.9) from running the code properly. However, the `bun` team has fixed my PR and it now works using the latest canary release. You can use this version by running `bun upgrade --canary`. I expect that once version 0.6.0 is released, the regular version will also work. Let me know if you run into any issues and thanks for the report!

**Comment by ornamentist** [comment, marked as answer]
Thank you for your hard work!
